### PR TITLE
feat: separate login hooks

### LINKED ISSUE #31: Login Basic UI Design

### PR DESCRIPTION
**Pull request by rizaardiyanto1412**
I confirmed this is works. You could test it with my login:

email: me@rizamaulana.com
password: 12345

### CODE REVIEW

**Review by alfonsusac — CHANGES_REQUESTED** [review]
move useLogin.jsx hook under the same file as LoginForm.jsx, below the component. 

The reason for this is that useLogin can only be used inside the LoginForm Component.

**Comment by rizaardiyanto1412** [comment]
@alfonsusac I put login func in lib/fetchApi.js as requested.

I confirm npm run build working fine.

File: src/lib/fetchAPI.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      const res = await fetch("https://eventmakers-api.fly.dev/auth/login", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          email,
+          password,
+        }),
+      });
```
Comment (alfonsusac) [inline src/lib/fetchAPI.js]: Hi, please only put this part into a `requestLogin()` function in /lib/fetchAPI.

The rest of the hook can be put inside `/LoginForm.jsx`

**Review by alfonsusac — CHANGES_REQUESTED** [review]

File: src/app/login/components/LoginForm.jsx [hunk 2]
```diff
@@ -1,38 +1,15 @@
...
-  }
+  const { loading, setEmail, setPassword, handleSubmitLogin } =
+    useRequestLogin();
 
   return (
-    <div className="flex w-full h-full">
-      <div className="bg-blue-500 h-[100vh] w-[50vw]"></div>
+    <div className="flex flex-col items-center justify-center m-auto h-[100vh] gap-10">
+      <h1 className="text-4xl">EVENTMAKER</h1>
+      <h1>LOGIN</h1>
```
Comment (alfonsusac) [inline src/app/login/components/LoginForm.jsx]: There can't be more than one `<h1>`. Please change the other one into `<h2>`
Login text should be bigger than Eventmaker.

Thankss

**Review by alfonsusac — CHANGES_REQUESTED** [review]

**Comment by rizaardiyanto1412** [comment]
@alfonsusac I adjusted as required.

I added an Image as dummy logo to give you idea, instead of putting text, I'm proposing to add logo instead

File: src/lib/fetchAPI.js [hunk 3]
```diff
@@ -1,62 +1,17 @@
...
+      email,
+      password,
+    }),
+  });
+
+  const data = await res.json();
+  return data;
+}
+
+export { requestLogin };
```
Comment (alfonsusac) [inline src/lib/fetchAPI.js:17]: Nice!

File: src/app/login/components/LoginForm.jsx [hunk 4]
```diff
@@ -1,15 +1,45 @@
...
+  const router = useRouter();
+  const [loading, setLoading] = useState(false);
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+
+  const handleSubmitLogin = async (e) => {
+    setLoading(true);
+    e.preventDefault();
+    try {
+      const { payload, token } = await requestLogin(email, password);
```
Comment (alfonsusac) [inline src/app/login/components/LoginForm.jsx:20]: NIce

**Review by alfonsusac — APPROVED** [review]

**Pull request merged**